DANS-KNAW/easy-convert-bag-to-deposit
# DD-682: use bag seq nr (of 2nd original versioned bag) for migration info

Fixes DD-682: use bag seq nr (of 2nd original versioned bag) for migration info

When applied it will
--------------------
* remove  `Bag-Sequence-Number: 2` from `bag-info.txt`
* use the value for migration-info request, the value defaults to `1`(one)
* 

Where should the reviewer @DANS-KNAW/easy start?
------------------------------------------------

How should this be manually tested?
-----------------------------------
For the short story, see https://drivenbydata.atlassian.net/browse/DD-682?focusedCommentId=44476
The long story below instructs how the zip was created.

* create a new deposit, 
* publish it and add some files to the root, next to original


* on easy-dtap: deploy a fresh fedora-to-bag, bag-to-deposit, validate-dans-bag
* on deasy
  * check versions with `rpm -qa |grep bag`

        dans.knaw.nl-easy-validate-dans-bag-1.4.5-SNAPSHOT...
        dans.knaw.nl-easy-convert-bag-to-deposit-1.0.18-SNAPSHOT...
        dans.knaw.nl-easy-fedora-to-bag-1.0.29-SNAPSHOT...

  * just once on a fresh VM

        mv /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/account-substitutes.csv /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/account-substitutes.csv.bu
        sudo mv /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/ThemathischeCollecties.csv ThemathischeCollecties-bu.csv
        sudo touch /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/ThemathischeCollecties.csv
        sudo chmod 777 /etc/opt/dans.knaw.nl/easy-convert-bag-to-deposit/

  * `rm -rf easy-*.log /vagrant/shared/btf-out/* /vagrant/shared/easy-fedora-to-bag-staging/*`
  * `easy-fedora-to-bag -d 'easy-dataset:17' -o /vagrant/shared/btf-out -f SIP original-versioned` # ID of a new manual deposit
  * replace the deasy DOI everywhere in both bags with a DOI known at `https://mds.test.datacite.org/` such as `10.80270/test-ztp-xjbk`
  * `bagit updatetagmanifests  /vagrant/shared/btf-out/<UUID>/<UUID>` for both bags (note that it would ignore metadata/easy-migration files created by bag-to-deposit)
  * keep a copy of these bags for the second run of bag-to-deposit
  * `easy-convert-bag-to-deposit -d /vagrant/shared/btf-out/ -t DOI -s FEDORA`
  * optionally: `easy-validate-dans-bag /vagrant/shared/btf-out/<UUID>/<UUID>` for both bags
* copy both bags created above: `easy-dtap/shared/btf-out/*` to `dd-dtap/shared/import-in`
* [x] import both bags on a fresh dev_archaeology-2021-11-07.box 

       dd-dans-deposit-to-dataverse import  --skip-validation /vagrant/shared/import-in /vagrant/shared/import-out/

* [x] fill and copy the migration-info DB from dar to deasy
  * on dar `curl X POST http://localhost:20230/datasets/actions/load-from-dataverse`
  * check the prestaged files with 

        curl -v -X GET 'http://localhost:20230/datasets/:persistentId/seq/1/basic-file-metas?persistentId=doi:10.80270/test-ztp-xjbk'

    and also with `seq/2`
  * on dar: `/opt/dans.knaw.nl/dd-migration-info/bin/export-migration-info.sh /vagrant/shared/miginfo.zip`
  * copy dd-dtap/shared/miginfo.zip to easy-dtap/shared
  * on deasy: `/opt/dans.knaw.nl/dd-migration-info/bin/import-migration-info.sh /vagrant/shared/miginfo.zip`

* [x] rerun `bag-to-deposit` (now with option `-p`) with the bags previously created with fedora-to-bag
  It produces valid bags with pre-staged.csv in metadata and only easy-migration/* in the data folder.

Related pull requests on github
-------------------------------
Requires and required by https://github.com/DANS-KNAW/easy-fedora-to-bag/pull/90
